freedomofpress/securedrop
# Ports configure logic to securedrop-admin script

## Status

Ready for review.

## Description of Changes

Fixes #1857.

The bash script at `install_files/ansible-base/configure` was already called via the `securedrop-admin`, providing a single entry point for Admins to memorize, and for the project documentation to cover. Since the bash script is not executed directly in the current docs, it's now been ported to a Python function inside the `securedrop-admin` management script. Functionality remains the same.

Related: #1409.

Does _not_ implement more user-friendly messaging as described in #1806—will open a separate PR for that.

## Testing

1. Boot into Tails with fresh persistence.
2. Run through the `./securedrop-admin setup` flow.
3. Run `./securedrop-admin sdconfig` and confirm the site-specific vars are created.
4. Run  `./securedrop-admin sdconfig` again and confirm you are _not_ prompted for the same vars (if you did not set a custom logo, you should see only one prompt, for the logo, which you can skip by pressing enter).
5.  Run  `./securedrop-admin sdconfig --force` and confirm you _are_ prompted for all the vars. Type some in differently, and confirm that you see the items as changed in the output.

## Deployment

Functionality remains the same as the old `./configure` script, so the major implications for deployment are adequate QA to confirm the changes work as described. 

## Checklist

Relying on manual testing using Tails.
